src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2253,7 +2235,7 @@ void Creature::setDeathState(DeathState s)
         if (m_formation && m_formation->GetLeader() == this)
             m_formation->FormationReset(true);
 
-        bool needsFalling = (IsFlying() || IsHovering()) && !IsUnderWater();
+        bool needsFalling = (IsFlying() || IsHovering()) && !IsUnderWater() && !IsSessile();
```

Comment:
I think this should check current root status (UNIT_STATE_ROOTED) instead of static flag (example: has root aura that is not removed on death)

Reply:
agreed